sympy/core/function.py

```diff
@@ -700,7 +701,20 @@ def _eval_nseries(self, x, n, logx):
                     raise PoleError("Cannot expand %s around 0" % (self))
                 series = term
                 fact = S.One
-                _x = Dummy('x')
+
+                # choose dummy name which isn't in self.args
+                def _modify(s):
```

Comment:
That's a good idea! I was wondering how one could accomplish the numbering.

But how about searching from the rhs?
```python
>>> def numbered_string_incr(s, start=0):
...     if not s:
...         return str(start)
...     i = len(s) - 1
...     while i != -1:
...         if not s[i].isdigit():
...             break
...         i -= 1
...     n = str(int(s[i + 1:] or start - 1) + 1)
...     return s[:i + 1] + n
>>> numbered_string_incr('')
0
>>> numbered_string_incr('', 1)
1
>>> numbered_string_incr('x')
x0
>>> numbered_string_incr('x', 1)
x1
>>> numbered_string_incr('x12', 1)
x13
>>> numbered_string_incr('x12', 14)
x13
>>> numbered_string_incr('x12_1')
x12_2
```
Such a function could replace `_uniquely_named_symbol`. Maybe it would be a good idea to add an underscore before the trailing number if the last character of the root is not already an underscore. I think that would look better when pretty printed.

Reply:
What do you think of adding this function to `sympy/core/symbols`? I can make another PR.

Reply:
I think I would promote `_uniquely_named_symbol` to public by removing its leading underscore and then make the default `number_string` (rename of `numbered_string_incr`) as you said.